jsapp/js/projects/projectsTable/projectBulkActions.tsx

```diff
@@ -3,30 +3,71 @@ import type {AssetResponse, ProjectViewAsset} from 'js/dataInterface';
 import Button from 'js/components/common/button';
 import actionsStyles from './projectActions.module.scss';
 import BulkDeletePrompt from './bulkActions/bulkDeletePrompt';
+import {userCan} from 'js/components/permissions/utils';
 
 interface ProjectBulkActionsProps {
   /** A list of selected assets for bulk operations. */
   assets: Array<AssetResponse | ProjectViewAsset>;
 }
 
+function userCanDeleteAssets(assets: Array<AssetResponse | ProjectViewAsset>) {
+  return assets.every((asset) => userCan('manage_asset', asset));
+}
```

Comment:
Can we wait til it's useful? I'm concerned it would add more confusion.

1. `(Array).every` makes this a pretty easy one-liner and this helper function does what it looks like without extra indirection.
1. Names are hard/misleading. What does `userCanPartially` do? If we add a nearby function called `userCanMany` or `userCanEvery`, they look related but they mean entirely different things.
1. If we're making a reusable function we need to consider edge cases that don't apply here yet, e.g. what should happen if you pass it an empty array – 'trivially true' or false? `(Array).every()` goes with "trivially true", but it doesn't matter because this component is only rendered when assets.length > 2.